### PR TITLE
fix(API): local .env

### PR DESCRIPTION
**Pull request by dfberry**
#343

1) Readme changes to help developer run API locally against Azure
2) package.json - change env file name to match code
3) config/index.ts - change code to match location of .env.local file

This change can go in before @glaucia86 migration to v4pm

### CODE REVIEW

File: packages/api/README.md [hunk 1]
```diff
@@ -17,17 +17,31 @@ If you want to run the API independently and locally, the following technologies
...
-```bash
-cd packages/api && npm install
-```
-Now you have all the dependencies installed for the API and can run
+1. Fork or clone the repository locally
+2. Provision Azure resources. 
 
-```bash
-npm start
-```
```
Comment (manekinekko) [inline packages/api/README.md:30]: these deleted steps are targeting the local dev setup (using the provided devcontainer setup). I don't think we should delete them. 

Developing against provisioned Azure PG instance could be added in a different section. WDYT?

Comment (diberry) [reply packages/api/README.md]: Reverted

**Comment by diberry** [comment]
@manekinekko Can you review and merge?

File: packages/api/README.md [hunk 2]
```diff
@@ -56,3 +56,38 @@ stripe trigger payment_intent.succeeded
 ```
 
 To test payments, you can use the [Stripe test cards](https://stripe.com/docs/testing#cards).
+
+## Local development with Azure resources
+
+To develop the Azure Functions app with the Azure databases, use the following steps.
+
+1. Fork the repository and open in Codespaces or Visual Studio dev container.
```
Comment (manekinekko) [inline packages/api/README.md:64]: note: while running the repo in Codespaces or devcontainer is technically the same thing, we only support Codespaces.

File: packages/api/README.md [hunk 3]
```diff
@@ -56,3 +56,38 @@ stripe trigger payment_intent.succeeded
...
+1. In a terminal at the root of the repository, use a terminal to login to Azure Developer CLI. Complete the steps.
```
Comment (manekinekko) [inline packages/api/README.md:65]:
```suggestion
1. At the root of the repository, use a terminal to login to Azure Developer CLI. Complete the steps.
```

File: packages/api/README.md [hunk 4]
```diff
@@ -56,3 +56,38 @@ stripe trigger payment_intent.succeeded
...
+  azd provision
+  ```
+
+1. Install the project dependencies.
+
+  ```bash
+  npm install
+  ```
+
+1. In a terminal at the `./packages/API` subfolder, create an `.env.local` file for the Azure Functions app. This creates an environment file with Azure resource values.
```
Comment (manekinekko) [inline packages/api/README.md:83]:
```suggestion
1. In a terminal at the `./packages/api` subfolder, create an `.env.local` file for the Azure Functions app. This creates an environment file with Azure resource values.
```

File: packages/api/README.md [hunk 5]
```diff
@@ -56,3 +56,38 @@ stripe trigger payment_intent.succeeded
...
+
+  ```bash
+  npm run env
+  ```
+
+1. Run the Functions app.
+
+  ```bash
+  npm run start
```
Comment (manekinekko) [inline packages/api/README.md:92]: We use npm start everywhere else.

```suggestion
  npm start
```

File: packages/api/config/index.ts [hunk 6]
```diff
@@ -18,7 +18,7 @@ export const getConfig: () => Promise<AppConfig> = async () => {
   // Load any ENV vars from local .env.local file
   if (process.env.NODE_ENV !== "production") {
     console.warn("Loading environment variables from root '.env.local' file. THIS SHOULD NOT BE USED IN PRODUCTION!");
-    dotenv.config({ path: path.resolve(process.cwd(), "../../.env.local") });
+    dotenv.config({ path: path.resolve(process.cwd(), ".env.local") });
```
Comment (manekinekko) [inline packages/api/config/index.ts:21]: I am afraid this change will break the API when running everything from the root of this monorepo. We need to think of a better flow to support both scenarios.

File: packages/api/README.md (hunk 2 above)
Comment (dfberry) [reply packages/api/README.md:64]: @manekinekko How would that be seen in the project to a developer new to the project?

File: packages/api/config/index.ts (hunk 6 above)
Comment (dfberry) [reply packages/api/config/index.ts]: @manekinekko the location in the api package.json for the 'env' script doesn't match the config/index.ts location. The [ed7f13b](https://github.com/Azure-Samples/contoso-real-estate/pull/344/commits/ed7f13b480a0b24a090cc8f2436521a7497dff71) commit fixes this.

**Comment by dfberry** [comment]
@manekinekko another review - this PR fixes the .env path and filename mismatch in package.json and /config/index.ts.

**Pull request closed without merging**